core/src/main/java/tc/oc/pgm/PGMPlugin.java

```diff
@@ -274,17 +274,40 @@ public void reloadConfig() {
     logger.setLevel(config.getLogLevel());
 
     for (String source : config.getMapSources()) {
-      final MapSourceFactory factory;
+      MapSourceFactory factory = null;
       try {
-        factory =
-            source.equalsIgnoreCase("default")
-                ? DefaultMapSourceFactory.INSTANCE
-                : new SystemMapSourceFactory(source);
+        if (source.equalsIgnoreCase("default"))
+          factory =
+              new GitMapSourceFactory(
+                  "https://github.com/KingOfSquares/DefaultMaps", logger, false);
+        if (source.startsWith("git;")) {
+          String gitSource = source.substring(4);
+          factory = new GitMapSourceFactory(gitSource, logger, true);
+        }
+        if (new File(source).exists()) factory = new SystemMapSourceFactory(source);
+      } catch (StringIndexOutOfBoundsException e) { // Git source parsing exception
```

Comment:
Add a method to `TextParser` to parse a git repository. `TextParser#parseGitRepository`

Reply:
Cant i just make an `GitRepository` object instead? Unsure how i would make a text parser for that since it has multiple strings as part of it `URIish`, `branch` and `folders`

Reply:
With the URIish from JGit it parses the url, password and username for me.